scripts/architecture_lint/baseline.py

```
@@ -0,0 +1,128 @@
+    allow_missing: bool,
+) -> dict[str, list[dict[str, object]] | None]:
+    run_git(root, "cat-file", "-e", f"{ref}^{{commit}}")
+    baselines: dict[str, list[dict[str, object]] | None] = {}
+    for rule in rules:
+        try:
+            raw = run_git(root, "show", f"{ref}:{rule.baseline_path.as_posix()}")
+        except GitDiscoveryError:
+            if allow_missing:
+                baselines[rule.id] = None
+                continue
+            raise ConfigurationError(f"baseline {rule.baseline_path} does not exist at {ref}")
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)